### PR TITLE
PODS-8479 - Refactor API transformations

### PR DESCRIPTION
**Pull request by LauraToddCodes**

### CODE REVIEW

File: app/transformations/ETMPToFrontEnd/API1831.scala
Comment (PavelVjalicin) [inline app/transformations/ETMPToFrontEnd/API1831.scala:1]: Why was this changed? Looks like simply repositioning variables around? If so, can we revert the changes to simplify.

File: app/transformations/ETMPToFrontEnd/API1833.scala [hunk 1]
```diff
@@ -41,48 +47,44 @@ object API1833 {
...
     ).reduce
 
-  val readsUnAuthorisedPaymentDetails: Reads[JsObject] = (
+  private val readsUnAuthorisedPaymentDetails: Reads[JsObject] = (
     readsUnAuthorisedPmtType1WithDynamicUAPaths and
       reqReads(pathUAUnAuthorisedPaymentDate, pathEtmpUnAuthorisedPaymentDetailsDateOfUnauthorisedPayment) and
       reqReads(pathUAUnAuthorisedPaymentValue, pathEtmpUnAuthorisedPaymentDetailsValueOfUnauthorisedPayment) and
-      optReadsDynamicPathStrTransform(dynamicPathUnAuthorisedPmtType2, pathEtmpUnAuthorisedPaymentDetailsUnAuthorisedPmtType2, unAuthorisedPmtType2Transform) and
+      optReadsDynamicPathStrTransform(
+        dynamicPathUnAuthorisedPmtType2, pathEtmpUnAuthorisedPaymentDetailsUnAuthorisedPmtType2, unAuthorisedPmtType2Transform) and
```
Comment (PavelVjalicin) [inline app/transformations/ETMPToFrontEnd/API1833.scala:79]: revert this change

File: app/transformations/Transformer.scala
Comment (PavelVjalicin) [inline app/transformations/Transformer.scala:1]: Why was code relocated here? Is this common transformations?

**Review by PavelVjalicin — CHANGES_REQUESTED** [review]

File: app/transformations/Transformer.scala
Comment (PavelVjalicin) [reply app/transformations/Transformer.scala:1]: If the Transformer code is not common, it should be put back

File: app/transformations/ETMPToFrontEnd/API1832.scala [hunk 2]
```diff
@@ -64,147 +63,109 @@ object API1832 {
...
-        pathUaBCEType.json.copyFrom(readsBCETypeEvent24) and
-        pathUaCrystallisedDateEvent24.json.copyFrom(pathEtmpTaxYearEndingDate.json.pick) and
-        pathUaFreeTextEvent24.json.copyFrom(pathEtmpFreeText.json.pick).orElse(doNothing) and
-        pathUaMarginalRate.json.copyFrom(pathEtmpMarginalRate.json.pick.flatMap(yesNoTransform(_, "taxed at marginal rate failed"))).orElse(doNothing) and
-        pathUaPayeReference.json.copyFrom(pathEtmpPayeReference.json.pick).orElse(doNothing)
-    ).reduce
-  }
+  lazy val readsEvent24PaymentDetails: Reads[JsObject] = (
+    reqReadsBoolTransform(pathUaMemberHoldProtection, pathEtmpMemberHoldProtectionEvent24, yesNoTransformToBoolean) and
+      optNestedReadsJsString(pathUaTypeOfProtectionEvent24, readsTypeOfProtectionEvent24) and
```
Comment (PavelVjalicin) [inline app/transformations/ETMPToFrontEnd/API1832.scala:113]: The yesNoTransformToBoolean is not equivalent. Can we re check other instances of yesNoTransformToBoolean use-case.

Comment (LauraToddCodes) [reply app/transformations/ETMPToFrontEnd/API1832.scala:113]: We weren't passing it into  `reqReadsBoolTransform` before so it needed to give a JsBoolean result rather than just a boolean. We had a lot of different methods transforming booleans so was trying to clean them up a bit.

Comment (PavelVjalicin) [reply app/transformations/ETMPToFrontEnd/API1832.scala:113]: I am okay with this as long as we have unit tests that test that this is not broken.

Comment (LauraToddCodes) [reply app/transformations/ETMPToFrontEnd/API1832.scala:113]: I think this would fail if it was broken - [https://github.com/hmrc/pension-scheme-event-reporting/blob/7f3a11a9df9594770d7938dbf6c5ab79fef9ffbb/test/utils/GeneratorAPI1832.scala#L490]( https://github.com/hmrc/pension-scheme-event-reporting/blob/7f3a11a9df9594770d7938dbf6c5ab79fef9ffbb/test/utils/GeneratorAPI1832.scala#L490)

**Review by PavelVjalicin — DISMISSED** [review]
Couldn't properly review, because of complexity of the changes.
I am okay with merging this into main, as long as we test in QA environment and make sure that QA team is comfortable with the changes.

**Review by PavelVjalicin — CHANGES_REQUESTED** [review]
Let's wait for bulk upload to be approved before merging, since bulk upload is also a big change.

**Review by PavelVjalicin — APPROVED** [review]
